src/components/logs/logsWebview.ts

```diff
@@ -0,0 +1,145 @@
+        const column = vscode.window.activeTextEditor ? vscode.window.activeTextEditor.viewColumn : undefined;
+
+        // If we already have a panel, show it.
+        const currentPanel = LogsPanel.currentPanels[resource];
+        if (currentPanel) {
+            currentPanel.setInfo(content, resource);
+            currentPanel.update();
+            currentPanel.panel.reveal(column);
+            return currentPanel;
+        }
+        const panel = vscode.window.createWebviewPanel(LogsPanel.viewType, "Kubernetes Logs", column || vscode.ViewColumn.One, {
+            enableScripts: true,
+
+            // And restrict the webview to only loading content from our extension's `media` directory.
+            localResourceRoots: [
+            ],
+        });
+
+        const result = new LogsPanel(panel, content, resource);
+        LogsPanel.currentPanels[resource] = result;
+        return result;
+    }
+
+    private constructor(
+        private readonly panel: vscode.WebviewPanel,
+        content: string,
+        resource: string
+    ) {
+        this.content = content;
+        this.resource = resource;
+
+        this.update();
+        this.panel.onDidDispose(() => this.dispose(), null, this.disposables);
+
+        this.panel.onDidChangeViewState((e: any) => {
+            if (this.panel.visible) {
+                this.update();
+            }
+        }, null, this.disposables);
+    }
+
+    public setInfo(content: string, resource: string) {
+        this.content = content;
+        this.resource = resource;
+        this.update();
+    }
+
+    public dispose() {
+        delete LogsPanel.currentPanels[this.resource];
+
+        this.panel.dispose();
+
+        while (this.disposables.length) {
+            const x = this.disposables.pop();
+            if (x) {
+                x.dispose();
+            }
+        }
+    }
+
+    private update() {
+        this.panel.title = `Kubernetes logs ${this.resource}`;
+        this.panel.webview.html = `
+        <!doctype html>
+        <html>
+        <head>
+            <meta charset="UTF-8">
+            <title>Kubernetes logs ${this.resource}</title>
+            <script
+                src="https://code.jquery.com/jquery-3.3.1.min.js"
+                integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8="
+                crossorigin="anonymous"></script>
+        </head>
+        <body>
+            <div style='position: absolute; top: 15px; left: 75%'>
+                <select id='mode' style='margin-bottom: 5px' onchange='eval()'>
+                    <option value='include'>Show Matches</option>
+                    <option value='exclude'>Hide Matches</option>
+                    <option value='after'>Show logs after match</option>
+                    <option value='before'>Show logs before match</option>
+                </select>
+                <input type='text' id='regexp' onkeyup='eval()' placeholder='Regular Expression Filter' size='25'/>
```

Comment:
I'm a bit doubtful about `eval`-ing on key up - for one thing, it's quite possible for the regex to be in an invalid state at that point, and even if it is valid it could well be transient (the user in the middle of typing a pattern), so we could be filtering a lot of logs for nothing.  I'd suggest an `Apply Filter` button, or, if you want it to be automatic, a delay where it applies the filter once the user has not typed anything for half a second or a second or something like that.

Reply:
In practice it seems to work pretty well. I do want it to be automatic.